packages/@aws-cdk/integ-runner/lib/runner/integ-test-case.ts

```diff
@@ -0,0 +1,230 @@
+import { TestCase, TestOptions } from '@aws-cdk/cloud-assembly-schema';
+import { ICdk, ListOptions } from 'cdk-cli-wrapper';
+import * as fs from 'fs-extra';
+import { IntegManifestReader } from './private/integ-manifest';
+
+const CDK_INTEG_STACK_PRAGMA = '/// !cdk-integ';
+const PRAGMA_PREFIX = 'pragma:';
+const SET_CONTEXT_PRAGMA_PREFIX = 'pragma:set-context:';
+const VERIFY_ASSET_HASHES = 'pragma:include-assets-hashes';
+const DISABLE_UPDATE_WORKFLOW = 'pragma:disable-update-workflow';
+const ENABLE_LOOKUPS_PRAGMA = 'pragma:enable-lookups';
+
+/**
+ * Represents an integration test
+ */
+export type TestCases = { [testName: string]: TestCase };
+
+/**
+ * Helper class for working with Integration tests
+ * This requires an `integ.json` file in the snapshot
+ * directory. For legacy test cases use LegacyIntegTestCases
+ */
+export class IntegTestCases {
+
+  /**
+   * Loads integ tests from a snapshot directory
+   */
+  public static fromPath(path: string): IntegTestCases {
+    const reader = IntegManifestReader.fromPath(path);
+    return new IntegTestCases(
+      reader.tests.enableLookups,
+      reader.tests.testCases,
+    );
+  }
+
+  constructor(
+    public readonly enableLookups: boolean,
+    public readonly testCases: TestCases,
+  ) {}
+
+  /**
+   * Returns a list of stacks that have stackUpdateWorkflow disabled
+   */
+  public getStacksWithoutUpdateWorkflow(): string[] {
+    const stacks: string[] = [];
+    for (const testCase of Object.values(this.testCases ?? {})) {
+      const update = testCase.stackUpdateWorkflow ?? true;
+      if (!update) {
+        stacks.push(...testCase.stacks);
+      }
+    }
+    return stacks;
```

Comment:
I find it easier to read like this:

```suggestion
    return Object.values(testCases)
      .filter(testCase => !testCase.stackUpdateWorkflow)
      .flatMap(testCase => testCase.stacks);
```